packages/cypress/src/add-commands/login.ts

```
@@ -374,10 +374,199 @@ function fillLoginForm(userCredentials: LoginCredentials) {
+          cy.get('button').contains('Next').click();
+
+          // Wait for the password form to appear
+          cy.get('input[name="password"]').should('be.visible');
+
+          // Fill in the password and submit
+          cy.get('input[name="password"]').type(userCredentials.password, {
+            log: false,
+          });
+          cy.get('button').contains('Submit').click();
+        }
+      );
+
+      // Set up fail handler for identity domain access
+      cy.on('fail', (err: Error) => {
+        // If we can't access identity domain, we're already on the app
+        if (
+          err.message.includes('Timed out retrying after') &&
+          err.message.includes('origin')
+        ) {
+          cy.log('Already on app domain, proceeding');
+          return false; // Prevent the error from failing the test
+        }
+        throw err; // Re-throw other errors
+      });
+
+      // Check if we're still on identity domain
+      cy.origin(
+        identityUrl,
+        {
+          args: { identityUrl },
+        },
+        () => {
+          // Set up fail handler for consent screen detection
+          cy.on('fail', (err: Error) => {
+            // If we can't find the consent screen, we're already logged in
+            if (
+              err.message.includes(
+                "Expected to find content: 'Access Requested' within the element"
+              )
+            ) {
+              cy.log('No consent screen found, proceeding to app');
+              return false; // Prevent the error from failing the test
+            }
+            throw err; // Re-throw other errors
+          });
+
+          // Try to find the consent screen
+          cy.get('h1')
+            .contains('Access Requested')
+            .should('exist')
+            .then(() => {
+              // Consent screen is present, handle it
+              cy.get('button').contains('Allow access').click();
+            });
+        }
+      );
```

I suppose this part is quite confusing. The problem is with the consent screen potentially not appearing, if for some reason the consent has been persisted in Ory.

It turned out to be impossible to reliably create conditional logic inside the `cy.origin` block in our specific build environment: 
https://docs.cypress.io/app/guides/conditional-testing#Element-existence
with login command always failing with always resulting in:
```
ReferenceError: _find is not defined
ReferenceError: _includes is not defined
ReferenceError: _indexOf is not defined
etc.
```
Our build environment aggressively transpiles JavaScript, making standard conditional checks inside `cy.origin()` impossible (causing `ReferenceError: _find is not defined`, `_includes is not defined`, etc.).

That is why in this proposal the command uses Cypress's global `cy.on('fail', ...)` event listener. In case the consent screen is not found, the test continues after failing silently.